tools/test_modules/m20710.pm

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env perl
+
+##
+## Author......: See docs/credits.txt
+## License.....: MIT
+##
+
+use strict;
+use warnings;
+
+use Digest::SHA  qw (sha256_hex);
+
+sub module_constraints { [[1, 31], [16, 16], [1, 31], [16, 16], [-1, -1]] }
```

Comment:
Almost there. Please change this restrictions to:

```
sub module_constraints { [[0, 256], [0, 256], [0, 55], [16, 16], [-1, -1]] }
```

I was wondering why the salt has be exactly length 16. In a generic mode, this should be more variable. Can you change the kernel? The pure kernel supports it already, it's just the optimized kernel.